### PR TITLE
Nukies Minigun is now more... Minigun-like

### PR DESCRIPTION
**Pull request by MajManatee**
# Document the changes in your pull request

Changes the standard burst to be 5rnds, and adds a 10rnd burst option.
What sort of LAME ASS MINIGUN fires in 3 round bursts? NOT THIS ONE ANYMORE BABY!!

# Changelog

:cl:  
rscadd: Minigun go BRRRRRRRRRRRRRRRT
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/ballistic/minigun.dm [hunk 1]
```diff
@@ -172,3 +174,25 @@
...
 	ammo_pack.attach_gun(user)
+
+/obj/item/gun/ballistic/minigunosprey/ui_action_click(mob/user, actiontype)
+	if(istype(actiontype, /datum/action/item_action/toggle_firemode))
+		burst_select()
+	else
+		..()
+
+/obj/item/gun/ballistic/minigunosprey/proc/burst_select()
+	var/mob/living/carbon/human/user = usr
```
Comment (tattax) [inline code/modules/projectiles/guns/ballistic/minigun.dm]:
```suggestion
	var/mob/living/carbon/user = usr
```

**Comment by VaelophisNyx** [comment]
why not add an emag mode that lets you click in a direction and, until you move, mag-dump?

File: code/modules/projectiles/guns/ballistic/minigun.dm [hunk 2]
```diff
@@ -172,3 +174,25 @@
...
+			select = 1
+			burst_size = initial(burst_size)
+			fire_delay = initial(fire_delay)
+			to_chat(user, span_notice("You switch to [burst_size]-rnd burst."))
+		if(1)
+			select = 0
+			burst_size = 10
+			fire_delay = initial(fire_delay)
+			to_chat(user, span_notice("You switch to [burst_size]-rnd burst. BRRRRRRRT."))
+	playsound(user, 'sound/weapons/empty.ogg', 100, 1)
```
Comment (tattax) [inline code/modules/projectiles/guns/ballistic/minigun.dm]:
```suggestion
		if(!select)
			select = TRUE
			burst_size = initial(burst_size)
			to_chat(user, span_notice("You switch to [burst_size]-rnd burst."))
		else
			select = FALSE
			burst_size = 10
			to_chat(user, span_notice("You switch to [burst_size]-rnd burst. BRRRRRRRT."))
	playsound(user, 'sound/weapons/empty.ogg', 100, TRUE)
```

File: code/modules/projectiles/guns/ballistic/minigun.dm [hunk 3]
```diff
@@ -172,3 +174,25 @@
...
+
+/obj/item/gun/ballistic/minigunosprey/ui_action_click(mob/user, actiontype)
+	if(istype(actiontype, /datum/action/item_action/toggle_firemode))
+		burst_select()
+	else
+		..()
+
+/obj/item/gun/ballistic/minigunosprey/proc/burst_select()
+	var/mob/living/carbon/human/user = usr
+	switch(select)
```
Comment (tattax) [inline code/modules/projectiles/guns/ballistic/minigun.dm]:
```suggestion

```

File: code/modules/projectiles/guns/ballistic/minigun.dm [hunk 4]
```diff
@@ -172,3 +174,25 @@
 /obj/item/gun/ballistic/minigunosprey/dropped(mob/living/user)
 	. = ..()
 	ammo_pack.attach_gun(user)
+
+/obj/item/gun/ballistic/minigunosprey/ui_action_click(mob/user, actiontype)
+	if(istype(actiontype, /datum/action/item_action/toggle_firemode))
+		burst_select()
+	else
+		..()
```
Comment (tattax) [inline code/modules/projectiles/guns/ballistic/minigun.dm]:
```suggestion
	. = ..()
```

Comment (tattax) [reply code/modules/projectiles/guns/ballistic/minigun.dm]: burst select always returns so you don't need an else

File: code/modules/projectiles/guns/ballistic/minigun.dm [hunk 5]
```diff
@@ -113,7 +113,8 @@
 	slot_flags = null
 	w_class = WEIGHT_CLASS_HUGE
 	materials = list()
-	burst_size = 3
+	burst_size = 5
+	var/select = 1
```
Comment (tattax) [inline code/modules/projectiles/guns/ballistic/minigun.dm]:
```suggestion
	var/select = TRUE
```

File: code/modules/projectiles/guns/ballistic/minigun.dm (hunk 1 above)
Comment (tattax) [reply code/modules/projectiles/guns/ballistic/minigun.dm]: also why is this defined in the first place it doesn't change anything cause you'll need to have the action to activate it anyways

**Comment by Hopekz** [comment]
> why not add an emag mode that lets you click in a direction and, until you move, mag-dump?

I don't think it would be consistent to be able to emag ballistic guns, so if this feature gets added I'd say toss it in to what the minigun is able to do from the start.

If this becomes a thing I'd like to mimic how miniguns in video games have a wind-up. Something like a heavy from Tf2 minigun windup before firing

**Comment by maxion12345** [comment]
oh good the nukie minigun needs a permit 
phew

**Review by JamieD1 — CHANGES_REQUESTED** [review]
Change spread to 30 and max burst to 5. Tested this and its a good compromise.

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**